### PR TITLE
Thread shift 0.17

### PR DESCRIPTION
**Pull request by rossabaker**
Builds on #1410, makes behavior like 0.16.  Specifically:

* runs routing and response task on `executorService` on blaze
* runs routing on `executorService` on servlets

### CODE REVIEW

File: server/src/main/scala/org/http4s/server/ServerBuilder.scala [hunk 1]
```diff
@@ -29,6 +29,10 @@ trait ServerBuilder {
   def withExecutorService(executorService: ExecutorService): Self =
     withExecutionContext(ExecutionContext.fromExecutorService(executorService))
 
+  @deprecated("Use withExecutionContext", "0.17.0")
```
Comment (rossabaker) [inline server/src/main/scala/org/http4s/server/ServerBuilder.scala]: This existed in 0.16 and was taken away and tripped up a user.  It tripped me up merging from 0.16.  This is just compiler-guided migration.

File: tomcat/src/test/scala/org/http4s/tomcat/TomcatServerSpec.scala [hunk 2]
```diff
@@ -2,6 +2,12 @@ package org.http4s
...
 package tomcat
 
-class TomcatServerSpec extends ServerAddressSpec {
+import org.apache.catalina.webresources.TomcatURLStreamHandlerFactory
+
+class TomcatServerSpec extends ServerSpec {
   val builder = TomcatBuilder
+
+  // Prevents us from loading jar and war URLs, but lets us
+  // run Tomcat twice in the same JVM.  This makes me grumpy.
```
Comment (ChristopherDavenport) [inline tomcat/src/test/scala/org/http4s/tomcat/TomcatServerSpec.scala]: Then it obviously needs a :(

File: servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala [hunk 3]
```diff
@@ -87,14 +87,14 @@ class Http4sServlet(service: HttpService,
...
-    val response = Task.start {
+    val response = Task.delay {
       try serviceFn(request)
         // Handle message failures coming out of the service as failed tasks
         .handleWith(serviceErrorHandler(request))
       catch
         // Handle message failures _thrown_ by the service, just in case
         serviceErrorHandler(request)
-    }(Strategy.fromExecutionContext(executionContext)).flatten
+    }.async(Strategy.fromExecutionContext(executionContext)).flatten
```
Comment (ChristopherDavenport) [inline servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala]: Should fix the thread location.

File: server/src/test/scala/org/http4s/server/ServerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    .mountService(HttpService {
+      case GET -> Root / "thread" / "routing" =>
+        val thread = Thread.currentThread.getName
+        Ok(thread)
+      
+      case GET -> Root / "thread" / "effect" =>
+        Task.delay(Thread.currentThread.getName).flatMap(Ok(_))
+    })
+    .start
+    .unsafeRun
```
Comment (ChristopherDavenport) [inline server/src/test/scala/org/http4s/server/ServerSpec.scala]: Not sure I like it this being in a lazy val, but it proves what we'd like for the tests, as well as abstracting concerns from the test sites.

File: server/src/test/scala/org/http4s/server/ServerSpec.scala [hunk 5]
```diff
@@ -1,45 +1,53 @@
...
 
   "A server" should {
+    val globalExecutorThreadPrefix = BuildInfo.scalaVersion match {
+      case v if v.startsWith("2.11.") => "ForkJoinPool-"
+      case _ => "scala-execution-context-global-"
+    }
+
     "route requests on the service executor" in {
-      get("/thread/routing").unsafeRun must startWith("scala-execution-context-global-")
+      println("foo")
```
Comment (bryce-anderson) [inline server/src/test/scala/org/http4s/server/ServerSpec.scala]: Are the `println`'s intentional?

Comment (rossabaker) [reply server/src/test/scala/org/http4s/server/ServerSpec.scala]: Not at all.

**Pull request merged**